src/CachePlugin.php

```diff
@@ -186,8 +209,11 @@ protected function isCacheable(ResponseInterface $response)
         if (!$this->config['respect_cache_headers']) {
             return true;
         }
-        if ($this->getCacheControlDirective($response, 'no-store') || $this->getCacheControlDirective($response, 'private')) {
-            return false;
+
+        foreach ($this->config['respect_response_cache_directives'] as $cacheDirective) {
+            if ($this->getCacheControlDirective($response, $cacheDirective)) {
```

Comment:
this means that if max-age is set to be respected and defined, we do not cache. i think you need a constant for NO_CACHE_FLAGS and array intersect that with respect_response_cache_directives